apps/extension/src/background/keyring/keyring.ts

```diff
@@ -77,9 +78,9 @@ export class KeyRing {
 
   constructor(
     protected readonly vaultService: VaultService,
+    protected readonly vaultStorage: VaultStorage,
```

Comment:
I agree with you that I'm not totally sure about passing vault service and vault storage together. It feels like it would leak the abstraction of having separate services, but I don't have a better idea. Maybe vault is just too big as a service and needs to be split into two services? I will think more about this, but for now I think this is good.

Reply:
What we maybe? could do is move more responsibility to handlers, as right now they just call single method. So handlers could:
- use vaults to get all the data needed
- pass data to services which just operate on the provided data
- persist data after services are done

This approach would be a bit closer to the DDD approach